src/services/dashboards/dashboard-customize/DashboardCustomizePage.vue

```diff
@@ -219,6 +219,10 @@ const handleSave = async () => {
 const handleChangeVariable = (variables: DashboardVariablesSchema['properties'], order?: string[]) => {
     variableState.variableProperties = variables;
     if (order) variableState.order = order;
+    variableState.order.forEach((d) => {
+        if (variables[d].use) return;
+        if (variableState.variableData[d]) delete variableState.variableData[d];
+    });
```

Comment:
I agree with @minseolee 's opinion! I prefer to create the copied object first.

```suggestion
   const _variableData = cloneDeep(variableState.variableData);
    variableState.order.forEach((d) => {
        if (variables[d].use) return;
        if (variableState.variableData[d]) delete _variableData[d];
    });
    variableState.variableData = _variableData;
```